### PR TITLE
chore: Upgrades `network_container` resource to auto-generated SDK

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Upgrades network_container resource to auto-generated SDK

Migration tests created.

Singular, plural data source and resource acceptance tests were being tested in the same test file, refactor to separate tests in the relevant test file.
 
Link to any related issue(s): CLOUDP-226085

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/networkcontainer/resource_network_container_migration_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+				ExternalProviders: mig.ExternalProviders(),
+				Config:            configAWS(projectName, orgID, cidrBlock, providerNameAws, "US_EAST_1"),
+				Check: resource.ComposeTestCheckFunc(
+					checkExists(resourceName, &container),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttr(resourceName, "provider_name", providerNameAws),
+					resource.TestCheckResourceAttrSet(resourceName, "provisioned"),
+				),
+			},
+			mig.TestStep(configAWS(projectName, orgID, cidrBlock, providerNameAws, "US_EAST_1")),
```
Comment (lantoli) [inline internal/service/networkcontainer/resource_network_container_migration_test.go]: I reccomend to extract to var section:   
config = configAWS(projectName, orgID, cidrBlock, providerNameAws, "US_EAST_1")

File: internal/service/networkcontainer/data_source_network_container.go [hunk 2]
```diff
@@ -74,70 +74,64 @@ func DataSource() *schema.Resource {
...
+	if err = d.Set("gcp_project_id", container.GetGcpProjectId()); err != nil {
 		return diag.FromErr(fmt.Errorf("error setting `gcp_project_id` for Network Container (%s): %s", d.Id(), err))
 	}
 
-	if err = d.Set("network_name", container.NetworkName); err != nil {
+	if err = d.Set("network_name", container.GetNetworkName()); err != nil {
 		return diag.FromErr(fmt.Errorf("error setting `network_name` for Network Container (%s): %s", d.Id(), err))
 	}
 
-	if err = d.Set("gcp_project_id", container.GCPProjectID); err != nil {
```
Comment (lantoli) [inline internal/service/networkcontainer/data_source_network_container.go:124]: was it twice before?

File: internal/service/networkcontainer/data_source_network_container_test.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: dataSourceConfigBasicAWS(projectName, orgID, cidrBlock, providerNameAws, "US_EAST_1"),
+				Check: resource.ComposeTestCheckFunc(
+					checkExists(resourceName, &container),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
```
Comment (lantoli) [inline internal/service/networkcontainer/data_source_network_container_test.go]: all checks should be checking dataSourceName not resourceName

File: internal/service/networkcontainer/data_source_network_container.go (hunk 2 above)
Comment (oarbusi) [reply internal/service/networkcontainer/data_source_network_container.go:124]: Exactly

File: internal/service/networkcontainer/resource_network_container_migration_test.go (hunk 1 above)
Comment (oarbusi) [reply internal/service/networkcontainer/resource_network_container_migration_test.go]: Done!

File: internal/service/networkcontainer/data_source_network_container_test.go (hunk 3 above)
Comment (oarbusi) [reply internal/service/networkcontainer/data_source_network_container_test.go]: Removed unecessary checks

File: internal/service/networkcontainer/data_source_network_container_test.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: dataSourceConfigBasicAzure(projectName, orgID, cidrBlock, providerNameAzure, "US_EAST_2"),
+				Check: resource.ComposeTestCheckFunc(
+					checkExists(resourceName, &container),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttr(resourceName, "provider_name", providerNameAzure),
```
Comment (lantoli) [inline internal/service/networkcontainer/data_source_network_container_test.go]: same, check dataSourceName

File: internal/service/networkcontainer/data_source_network_container_test.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+		PreCheck:                 func() { acc.PreCheck(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: dataSourceConfigBasicGCP(projectName, orgID, gcpCidrBlock, providerNameGCP),
+				Check: resource.ComposeTestCheckFunc(
+					checkExists(resourceName, &container),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttr(resourceName, "provider_name", providerNameGCP),
```
Comment (lantoli) [inline internal/service/networkcontainer/data_source_network_container_test.go]: same

File: internal/service/networkcontainer/data_source_network_containers_test.go [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheck(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		Steps: []resource.TestStep{
+			{
+				Config: dataSourcePluralConfigBasicAWS(projectName, orgID, cidrBlock, providerNameAws, "US_EAST_1"),
+				Check: resource.ComposeTestCheckFunc(
+					checkExists(resourceName, &container),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttr(resourceName, "provider_name", providerNameAws),
```
Comment (lantoli) [inline internal/service/networkcontainer/data_source_network_containers_test.go]: i don't think we need to check resource in plural ds

File: internal/service/networkcontainer/data_source_network_containers_test.go [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+				),
+			},
+		},
+	})
+}
+
+func dataSourcePluralConfigBasicAWS(projectName, orgID, cidrBlock, providerName, region string) string {
+	return fmt.Sprintf(`
+		resource "mongodbatlas_project" "test" {
+			name   = "%s"
```
Comment (lantoli) [inline internal/service/networkcontainer/data_source_network_containers_test.go:84]: "%s" is better %q

File: internal/service/networkcontainer/data_source_network_containers_test.go [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+		}
+		
+		resource "mongodbatlas_network_container" "test" {
+			project_id   		 = mongodbatlas_project.test.id
+			atlas_cidr_block     = "%s"
+			provider_name		 = "%s"
+		}
+
+		data "mongodbatlas_network_containers" "test" {
+			project_id = "${mongodbatlas_network_container.test.project_id}"
```
Comment (lantoli) [inline internal/service/networkcontainer/data_source_network_containers_test.go:138]: "${mongodbatlas_network_container.test.project_id}"   =>  mongodbatlas_network_container.test.project_id

Comment (oarbusi) [reply internal/service/networkcontainer/data_source_network_containers_test.go:138]: Done

File: internal/service/networkcontainer/resource_network_container.go [hunk 9]
```diff
@@ -103,63 +103,61 @@ func Resource() *schema.Resource {
...
-	containerRequest := &matlas.Container{
-		AtlasCIDRBlock: d.Get("atlas_cidr_block").(string),
-		ProviderName:   providerName,
+	atlasCidrBlock := d.Get("atlas_cidr_block").(string)
+	containerRequest := &admin.CloudProviderContainer{
+		AtlasCidrBlock: &atlasCidrBlock,
+		ProviderName:   &providerName,
 	}
 
 	if providerName == "AWS" {
```
Comment (lantoli) [inline internal/service/networkcontainer/resource_network_container.go]: already exists:
constant.AZURE
constant.AWS

you can add GCP if needed

File: internal/service/networkcontainer/resource_network_container_test.go [hunk 10]
```diff
@@ -228,7 +192,7 @@ func testAccCheckMongoDBAtlasNetworkContainerImportStateIDFunc(resourceName stri
 	}
 }
 
-func testAccCheckMongoDBAtlasNetworkContainerExists(resourceName string, container *matlas.Container) resource.TestCheckFunc {
+func checkExists(resourceName string, container *admin.CloudProviderContainer) resource.TestCheckFunc {
```
Comment (lantoli) [inline internal/service/networkcontainer/resource_network_container_test.go]: i would remove container from checkExists

**Review by lantoli — APPROVED** [review]

**Pull request merged**